src/crab/wto_nesting.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Prevail Verifier contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+// Bournacle, "Efficient chaotic iteration strategies with widenings", 1993
+// uses the notation w(c) to refer to the set of heads of the nested components
+// containing a vertex c.  This class holds such a set of heads.  The table
+// mapping c to w(c) is stored outside the class, in wto_t._nesting.
+class wto_nesting_t final {
+    // To optimize insertion performance, the list of heads is stored in reverse
+    // order, i.e., from innermost to outermost cycle.
+    std::vector<label_t> _heads;
+
+  public:
+    wto_nesting_t(std::vector<label_t>& heads) : _heads(std::move(heads)) {}
```

Comment:
IIUC think this constructor is designed to accept only vectors created specifically for it. So IMO it should either accept an rvalue reference (letting the compiler verify that on the caller side) or accept by value, requiring the caller to understand that they should pass it an rvalue. Calling a move-constructor on a parameter passed by reference might be surprising to the caller.

Reply:
Done.  moved the std::move() out to the caller.